src/msg/async/ProtocolV1.cc

```diff
@@ -337,7 +341,7 @@ void ProtocolV1::write_event() {
       connection->write_lock.unlock();
 
       // send_message or requeue messages may not encode message
-      if (!data.length()) {
+      if (!data.length() || out_entry.is_prepared) {
```

Comment:
Oops, shouldn't `out_entry.is_prepared` be negated here? My understanding is we would call `prepare_send_message()` **twice** if the `can_fast_prepare && f` check in `send_message()` has failed (returned `false`).

Am I missing something?